transactor/src/main/scala/io/mediachain/copycat/S3BackingStore.scala

```diff
@@ -0,0 +1,252 @@
+package io.mediachain.copycat
+
+import java.util.Properties
+import java.util.concurrent.{ExecutorService, Executors, TimeUnit}
+import java.io.{File, FileOutputStream}
+import com.amazonaws.AmazonClientException
+import com.amazonaws.auth.{AWSCredentials, BasicAWSCredentials}
+import com.amazonaws.services.s3.AmazonS3Client
+import org.slf4j.{Logger, LoggerFactory}
+import scala.concurrent.Await
+import scala.concurrent.duration.Duration
+import scala.collection.JavaConversions._
+import scala.util.{Try, Success, Failure}
+import cats.data.Xor
+import io.mediachain.protocol.Transactor._
+import io.mediachain.protocol.Datastore._
+import io.mediachain.protocol.CborSerialization._
+import io.mediachain.copycat.Client.{ClientState, ClientStateListener}
+import io.mediachain.datastore.DynamoDatastore
+
+class S3BackingStore(config: S3BackingStore.Config)
+extends JournalListener with ClientStateListener with AutoCloseable {
+  import S3BackingStore.multihashRef
+  private val logger = LoggerFactory.getLogger(classOf[S3BackingStore])
+  private val exec = Executors.newScheduledThreadPool(Math.max(4, Runtime.getRuntime.availableProcessors))
+  private val datastore = new DynamoDatastore(config.dynamo)
+  private val s3 = new AmazonS3Client(config.awscreds)
+  private val s3bucket = config.s3bucket
+  private val client = Client.build(config.sslConfig)
+  private var cluster: Option[List[String]] = None
+  private var blocks: Set[Reference] = Set()
+  
+  client.addStateListener(this)
+
+  def connect(addrs: List[String]) {
+    cluster = Some(addrs)
+    client.connect(addrs)
+    client.listen(this)
+  }
+  
+  def close() {
+    exec.shutdownNow()
```

Comment:
It is kind of a safe operation -- a restart will pick up where you stopped and you can't have incomplete puts in S3.
Also,  if you are calling close here you are probably exiting the entire program and you won't it to stop asap.